Error handling wrapper

pulkit21 opened the pull request.

pulkit21 committed:
base handeling

pulkit21 committed:
Implemented base log for showing errors

devilankur18 commented on apps/factory/src/components/prompt_version.tsx:
```diff
@@ -210,38 +210,42 @@ function PromptVersion({
-      {
-        onSuccess() {
-          setIsRunning(false);
+    try {
```
@pulkit21 this try catch is wrong, there is on Error and onSettled callback. that should be used.

devilankur18 commented on apps/factory/src/services/vendors/base_vendor.ts:
```diff
@@ -78,24 +82,30 @@ export async function fetchWithRetry(
   maxRetries: number,
   retryDelay: number,
-) {
+): Promise<{ data: any; error: null } | { data: null; error: any }> {
```
Make this a datatype, so can be used everywhere

devilankur18 commented on apps/factory/src/services/vendors/base_vendor.ts:
```diff
@@ -104,7 +114,55 @@ export async function fetchWithRetry(
+      const clientError = {
+        code: "CLIENT_ERROR",
+        message: "Client error occurred.",
+        vendorCode,
```
create a global object with error codes and messages, so we can referece it all the places. this codes will be hiddne inside and will be difficult to maintain.

pulkit21 committed:
Updated the errorhandling in seperate file and made the requested changes

pulkit21 committed:
Added the error handling for all the models/provides

pulkit21 committed:
Refactor code and added types
Things to Do: fix completion in prompt_output, prompt_template_view and other type issues

pulkit21 committed:
fixed the output screen for both cube n template

devilankur18 committed:
refactored and build fixes
